recipes/openusd/all/conanfile.py

```diff
@@ -0,0 +1,526 @@
+#    29 | #include <Ptexture.h>
+
+        "enable_ptex_support": False,
+        "enable_openvdb_support": False,
+        "build_renderman_plugin": False,
+        # FIXME:
+#         /root/.conan2/p/b/openu0d9487be679c5/b/src/pxr/usd/plugin/usdAbc/alembicUtil.h:38:10: fatal error: Alembic/Abc/ICompoundProperty.h: No such file or directory
+#    38 | #include <Alembic/Abc/ICompoundProperty.h>
+
+        "build_alembic_plugin": False,
+        "enable_hdf5_support": False,
+        # FIXME:
+#         /root/.conan2/p/b/openub6914422cb70b/b/src/pxr/usd/plugin/usdDraco/attributeDescriptor.h:34:10: fatal error: draco/attributes/geometry_attribute.h: No such file or directory
+#    34 | #include <draco/attributes/geometry_attribute.h>
+
+        "build_draco_plugin": False,
+        "enable_osl_support": False,
+        "build_animx_tests": False,
+        "enable_python_support": True,
+    }
+
+    short_paths = True
+
+    @property   
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "10",
+            "clang": "7",
+            "gcc": "7",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.options.build_usd_imaging and not self.options.build_imaging:
+            self.options.build_usd_imaging = False
+        self.options.build_gpu_support = self.options.enable_gl_support or is_apple_os(self) or self.options.enable_vulkan_support
+        self.options["opensubdiv"].with_opengl = self.options.enable_gl_support
+        if is_apple_os(self):
+            self.options["opensubdiv"].with_metal = True
+        if self.options.build_usdview:
+            if self.options.build_imaging:
+                self.options.build_usd_imaging = False
+        if self.options.build_usdview:
+            if not self.options.build_usd_imaging:
+                self.options.build_usdview = False
+            elif not self.options.enable_python_support:
+                self.options.build_usdview = False
+            elif not self.options.build_gpu_support:
+                self.options.build_usdview = False
+        if self.options.build_embree_plugin:
+            if not self.options.build_imaging:
+                self.options.build_embree_plugin = False
+            elif not self.options.build_gpu_support:
+                self.options.build_embree_plugin = False
+        if self.options.build_renderman_plugin:
+            if not self.options.build_imaging:
+                self.options.build_renderman_plugin = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if self.settings.os == "Windows":
+             self.options["openssl"].shared = True
```

Comment:
you will need to validate this option in the validate() method if this is a hard requirement, as users can override recipe choices

Reply:
Check here for an example of a nice message to give to users https://github.com/conan-io/conan-center-index/pull/24480/files/985651bff18949278eb0da17f8ee32710c20c377#diff-d0428b639a2f228893f252d9582c8726cb6d7dffb159fea13ddbac1ac3772f65R119 (plus an explanation, if necessary)

Reply:
Hi, I have removed this part.
I leave this PR in draft as it is not yet ready for review.
Sorry.

Reply:
Ready for review.